keep-ui/app/providers/provider-form.tsx

```
@@ -96,52 +100,72 @@ function base64urlencode(a) {
   return btoa(str).replace(/\+/g, "-").replace(/\//g, "_").replace(/=+$/, "");
 }
 
-const DictInput = ({ name, value, onChange, error }) => {
-  const handleEntryChange = (index, field, newValue) => {
-    const newEntries = value.map((entry, i) =>
-      i === index ? { ...entry, [field]: newValue } : entry
+function getRequiredConfigs(config: Provider["config"]): Provider["config"] {
+  return Object.entries(config)
+    .filter(([_, config]) => config.required && !config.config_main_group)
+    .reduce((acc, [key, value]) => ({ ...acc, [key]: value }), {});
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)